src/AggregateFunctions/AggregateFunctionMinMaxAny.h

```diff
@@ -960,6 +979,62 @@ struct AggregateFunctionAnyData : Data
 #endif
 };
 
+/** Implements aggregate function Any (first_value) with Respect Nulls modifier and nullable result.
+  */
+template <typename Data>
+struct AggregateFunctionAnyDataRespectNulls : AggregateFunctionAnyData<Data>
+{
+    using Self = AggregateFunctionAnyDataRespectNulls;
+    static constexpr bool is_nullable = true;
+    static constexpr bool is_any = true;
+    static constexpr bool is_respect_nulls = true;
+
+    static const char * name() { return "any_respect_nulls"; }
+
+    bool is_null = true;
+
+    bool changeIfBetter(const IColumn & column, size_t row_num, Arena * arena)
+    {
+        is_null &= Data::has();
+        return this->changeFirstTime(column, row_num, arena);
+    }
+
+    bool changeIfBetter(const Self & to, Arena * arena)
+    {
+        is_null &= Data::has();
+        return this->changeFirstTime(to, arena);
+    }
+
+    void addManyDefaults(const IColumn & column, size_t /*length*/, Arena * arena) { this->changeIfBetter(column, 0, arena); }
+
+    void insertResultInto(IColumn & to) const
+    {
+        if (is_null)
+        {
+            to.insertDefault();
+        }
+        else
+        {
+            ColumnNullable & col = typeid_cast<ColumnNullable &>(to);
+            col.getNullMapColumn().insertDefault();
+            this->Data::insertResultInto(col.getNestedColumn());
+        }
+    }
+};
+
+/** Implements aggregate function Any (first_value) with Ignore Nulls modifier,
+    result is not nullable.
+  */
+template <typename Data>
+struct AggregateFunctionAnyDataIgnoreNulls : AggregateFunctionAnyData<Data>
```

Comment:
Am I missing something or why do we need a separate `AggregateFunctionAnyDataIgnoreNulls` class if it seems like the behavior is the same as in the `AggregateFunctionAnyData` class?

Reply:
> Am I missing something or why do we need a separate `AggregateFunctionAnyDataIgnoreNulls` class if it seems like the behavior is the same as in the `AggregateFunctionAnyData` class?

Just thought it would be clearer and for code consistency.